pkg/cryptoutils/certificate.go

```diff
@@ -76,6 +76,35 @@ func UnmarshalCertificatesFromPEM(pemBytes []byte) ([]*x509.Certificate, error)
 	return result, nil
 }
 
+// UnmarshalCertificatesFromPEMLimited extracts one or more X509 certificates from the provided
+// byte slice, which is assumed to be in PEM-encoded format. Fails after a specified
+// number of iterations.
+func UnmarshalCertificatesFromPEMLimited(pemBytes []byte, iterations int) ([]*x509.Certificate, error) {
```

Comment:
Nice! Should we suggest a safe/reasonable limit?

Reply:
Added a suggestion of 10! It'd be quite unexpected to see a chain longer than that.